drivers/sensor/vl53l1x/vl53l1.c

```diff
@@ -0,0 +1,482 @@
+{
+	struct vl53l1x_data *drv_data = dev->data;
+	VL53L1_Error ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_DISTANCE);
+
+	if (!drv_data->initialized) {
+		int init_ret = vl53l1x_initialize(dev);
+
+		if (init_ret < 0) {
+			return init_ret;
+		}
+	}
+
+	/* Will immediately stop current measurement */
+	ret = VL53L1_StopMeasurement(&drv_data->vl53l1x);
+	if (ret != VL53L1_ERROR_NONE) {
+		LOG_ERR("VL53L1_StopMeasurement return error (%d)", ret);
+		return -EINVAL;
+	}
+
+#ifdef CONFIG_VL53L1X_INTERRUPT_MODE
+	vl53l1x_enable_interrupt(drv_data->dev, true);
+#endif
+
+	ret = VL53L1_StartMeasurement(&drv_data->vl53l1x);
+	if (ret != VL53L1_ERROR_NONE) {
+		LOG_ERR("[%s] VL53L1_StartMeasurement return error (%d)", dev->name, ret);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int vl53l1x_channel_get(const struct device *dev,
+							   enum sensor_channel chan,
+							   struct sensor_value *val)
+{
+	struct vl53l1x_data *drv_data = dev->data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_DISTANCE);
+
+#ifndef CONFIG_VL53L1X_INTERRUPT_MODE
+	/* Using driver poling mode */
+	ret = VL53L1_WaitMeasurementDataReady(&drv_data->vl53l1x);
+	if (ret != VL53L1_ERROR_NONE) {
+		LOG_ERR("[%s] VL53L1_WaitMeasurementDataReady return error (%d)", dev->name, ret);
+		return -EINVAL;
+	}
+	vl53l1x_read_sensor(drv_data);
+#endif
+
+	val->val1 = (int32_t)(drv_data->data.RangeMilliMeter);
+	/* RangeFractionalPart not implemented in API */
+	val->val2 = 0;
+
+	return 0;
+}
+
+static int vl53l1x_attr_get(const struct device *dev,
+							enum sensor_channel chan,
+							enum sensor_attribute attr,
+							struct sensor_value *val)
+{
+	__ASSERT_NO_MSG((chan == SENSOR_CHAN_DISTANCE) &&
+					((attr == SENSOR_ATTR_CONFIGURATION) ||
+					(attr == SENSOR_ATTR_CALIB_TARGET)));
+
+	if (attr == SENSOR_ATTR_CONFIGURATION) {
+		vl53l1x_get_mode(dev, val);
+	} else if (attr == SENSOR_ATTR_CALIB_TARGET) {
+		vl53l1x_get_roi(dev, val);
+	}
+
+	return 0;
+}
+
+static int vl53l1x_attr_set(const struct device *dev,
+							enum sensor_channel chan,
+							enum sensor_attribute attr,
+							const struct sensor_value *val)
+{
+	__ASSERT_NO_MSG((chan == SENSOR_CHAN_DISTANCE) &&
+					((attr == SENSOR_ATTR_CONFIGURATION) ||
+					(attr == SENSOR_ATTR_CALIB_TARGET)));
+
+	if (attr == SENSOR_ATTR_CONFIGURATION) {
+		vl53l1x_set_mode(dev, val);
+	} else if (attr == SENSOR_ATTR_CALIB_TARGET) {
+		vl53l1x_set_roi(dev, val);
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api vl53l1x_api_funcs = {
+	.sample_fetch = vl53l1x_sample_fetch,
+	.channel_get = vl53l1x_channel_get,
+	.attr_get = vl53l1x_attr_get,
+	.attr_set = vl53l1x_attr_set,
+};
+
+static int vl53l1x_init(const struct device *dev)
+{
+	int ret = 0;
+	struct vl53l1x_data *drv_data = dev->data;
+	const struct vl53l1x_config *const config = dev->config;
+
+	/* Initialize the HAL peripheral with the default sensor address,
+	 * ie. the address on power up
+	 */
+	drv_data->vl53l1x.i2c_slave_address = VL53L1X_INITIAL_ADDR;
```

Comment:
You should be able to get the client address from the device tree.  I.e., `reg = <0x29>`.  Putting in the dts but then ignoring it and using a hard-coded value doesn't seem right.

Reply:
Yes my mistake. Forgot to replace.